src/app/services/auth.service.ts

```diff
@@ -48,9 +57,20 @@ export class AuthService {
 
   public async completeAuthentication() {
     this.user = await this.manager.signinRedirectCallback();
+    this.Backenduser = await this.getBackendUser();
     this._authNavStatusSource.next(this.isAuthenticated());
   }
 
+  public async getBackendUser(): Promise<BackendUser>{
+      return this.http.get<BackendUser>(`${API_CONFIG.url}User`).toPromise();
+  }
+  public getCurrentRole(): Role{
+    if(this.Backenduser == null){
```

Comment:
You should also check if the backendUser has a role. Otherwise this could cause null pointer exceptions